src/test/util/event.rs

```diff
@@ -464,74 +458,7 @@ impl CommandEventHandler for EventHandler {
     }
 }
 
-#[derive(Debug)]
-pub(crate) struct EventSubscriber<'a> {
-    /// A reference to the handler this subscriber is receiving events from.
-    /// Stored here to ensure this subscriber cannot outlive the handler that is generating its
-    /// events.
-    _handler: &'a EventHandler,
-    receiver: tokio::sync::broadcast::Receiver<Event>,
-}
-
-impl EventSubscriber<'_> {
```

Comment:
this type has been made generic over the handler and event types and moved into `src/test/util/subscriber.rs` so we can reuse it for tracing events as well. 
`wait_for_successful_command_execution` is only meaningful for an `EventSubscriber<'_, EventHandler, Event>` though so I left that method here.